File: commons/java/test/lib/AnswerTest.java
package lib;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class AnswerTest {

    @Test
    void testSuccess() {
        Answer.part1(1, 1);
        Answer.part2("value", "value");
    }

    @Test
    void testFailure() {
        Assertions.assertThrows(RuntimeException.class, () -> Answer.part1(1, 2));
    }
}

File: commons/rust/tests/answer_test.rs
use aoc_lib::answer;

#[test]
fn test_correct() {
    answer::part1(2, 2);
    answer::part2("abc", "abc");
}

#[test]
#[should_panic]
fn test_incorrect() {
    answer::part1(2, -2);
}
